framework/log/Dispatcher.php

```diff
@@ -80,7 +80,7 @@ public function __construct($config = [])
     {
         // ensure logger gets set before any other config option
         if (isset($config['logger'])) {
-            $this->setLogger($config['logger']);
+            $this->setLogger(Yii::createObject($config['logger']));
```

Comment:
Well with your fix you can set object instance via `setLogger()`, but not via constructor config. I don't think this is the way to go. Better do is_array check and don't touch the tests.

Though you may wait until Yii team member makes a comment. :)